### PR TITLE
Add extract_cells_by_type

### PR DESCRIPTION
**Pull request by banesullivan**
This is a filter to help resolve https://github.com/pyvista/pyvista/issues/3738 and issues like it.

`extract_cells_by_type()` is a new dataset filter that wraps `vtkExtractCellsByType` to clean meshes with specific cell types. This is especially useful when needing to remove vertex cells from a mesh that is a surface mesh.

Useability is my last concern, as this requires users to know exactly what cell type to extract and how to look up its integer value. Some work is needed here to streamline access to the cell-type integers.

### CODE REVIEW

**Comment by banesullivan** [comment]
Aha! Thanks for pointing me to `CellType`, @adeak! Totally missed that while implementing this before my coffee this morning. Will update to leverage that

File: pyvista/core/filters/data_set.py [hunk 1]
```diff
@@ -5535,6 +5535,56 @@ def separate_cells(self):
...
+
+        .. warning::
+            Unlike the filter vtkExtractCells which always produces
+            vtkUnstructuredGrid output, this filter produces the same output
+            type as input type (i.e., it is a vtkDataSetAlgorithm). Also,
+            vtkExtractCells extracts cells based on their ids.
+
+        Parameters
+        ----------
+        cell_types : bool | int | list[int]
```
Comment (MatthewFlamm) [inline pyvista/core/filters/data_set.py:5565]: What is the use case for `cell_types=True` here?  Won't this always return the same data object back if it has cells?  My reading of `vtkExtractCellsByType` is that `AddAllCellTypes` could be useful in conjunction with `RemoveCellType` to select all types but one, for example.  If this is correct, I'd suggest dropping this usage from this method.  Instead, we could use a convenience method for generating a list of all cell types using `pyvista.CellType`.  For example, can we do something like:

```py
cell_types = pv.CellType.all_cells(exclude=pv.CellType.QUADRATIC_EDGE)
mesh.extract_cells_by_type(cell_types)
```

Bonus points if the design in `CellType` could allow us to distinguish between different flavors based on some criteria, e.g. dimensionality  `pv.CellType.all_edge_cells().  This is out of scope for this PR, but I'm trying to provide some argument for the design of this method to stay out of the game of trying to configure cell types via input.

Comment (akaszynski) [reply pyvista/core/filters/data_set.py]: Resolved this in [220f426](https://github.com/pyvista/pyvista/pull/4204/commits/220f4266f267a89d2c6e0b1794fb1f209f7ab598).

**Comment by akaszynski** [comment]
> Useability is my last concern, as this requires users to know exactly what cell type to extract and how to look up its integer value. Some work is needed here to streamline access to the cell-type integers.

I think we've simplified this with an example using [pyvista.CellType](https://docs.pyvista.org/version/stable/api/utilities/_autosummary/pyvista.CellType.html) as suggested by @adeak.

**Review by akaszynski — APPROVED** [review]

**Pull request merged**